libs/nucs/src/token.rs

```diff
@@ -0,0 +1,308 @@
+use crate::policy::Policy;
+use chrono::{DateTime, Utc};
+use hex::FromHexError;
+use serde::Deserialize;
+use serde_with::DeserializeFromStr;
+use std::{fmt, str::FromStr};
+
+/// A JSON object.
+pub type JsonObject = serde_json::Map<String, serde_json::Value>;
+
+/// A Nillion NUC token.
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+pub struct NucToken {
+    /// The token issuer.
+    #[serde(rename = "iss")]
+    pub issuer: Did,
+
+    /// The token issuer.
+    #[serde(rename = "aud")]
+    pub audience: Did,
+
+    /// The token subject.
+    #[serde(rename = "sub")]
+    pub subject: Did,
+
+    /// The first timestamp at which this token is valid.
+    #[serde(rename = "nbf", default)]
+    pub not_before: Option<DateTime<Utc>>,
```

Comment:
Shouldn't this be numerical unix timestamp?

Reply:
True!